Studies/python/SignalSelectionModel.py

```diff
@@ -0,0 +1,90 @@
+import tensorflow as tf
+import numpy as np
+import json
+
+from tensorflow.keras.layers import Layer, Dense, Dropout, LSTM, TimeDistributed, Concatenate
+from tensorflow.keras.models import Model
+
+class StdLayer(Layer):
+    def __init__(self, file_name, var_pos, n_sigmas):
+        with open(file_name) as json_file:
+            data_json = json.load(json_file)
+        n_vars = len(var_pos)
+        self.vars_mean = [0] * n_vars
+        self.vars_std = [0] * n_vars
+        self.vars_apply = [False] * n_vars
+        for var, ms in data_json.items():
+            pos = var_pos[var]
+            self.vars_mean[pos] = ms['mean']
+            self.vars_std[pos] = ms['std']
+            self.vars_apply[pos] = True
+        self.vars_mean = tf.constant(self.vars_mean, dtype=tf.float32)
+        self.vars_std = tf.constant(self.vars_std, dtype=tf.float32)
+        self.vars_apply = tf.constant(self.vars_apply, dtype=tf.bool)
+        self.n_sigmas = n_sigmas
+
+        super(StdLayer, self).__init__()
+
+    def build(self, input_shape):
+        super(StdLayer, self).build(input_shape)
+
+    def call(self, X):
+        Y = tf.clip_by_value(( X - self.vars_mean ) / self.vars_std, -self.n_sigmas, self.n_sigmas)
+        return tf.where(self.vars_apply, Y, X)
+
+class ScaleLayer(Layer):
+    def __init__(self, file_name, var_pos, interval_to_scale):
+        with open(file_name) as json_file:
+            data_json = json.load(json_file)
+        self.a = interval_to_scale[0]
+        self.b = interval_to_scale[1]
+        n_vars = len(var_pos)
+        self.vars_min = [0] * n_vars
+        self.vars_max = [0] * n_vars
+        self.vars_apply = [False] * n_vars
+        for var, mm in data_json.items():
+            pos = var_pos[var]
+            self.vars_min[pos] = mm['min']
+            self.vars_max[pos] = mm['max']
+            self.vars_apply[pos] = True
+        self.vars_min = tf.constant(self.vars_min, dtype=tf.float32)
+        self.vars_max = tf.constant(self.vars_max, dtype=tf.float32)
+        self.vars_apply = tf.constant(self.vars_apply, dtype=tf.bool)
+
+        super(ScaleLayer, self).__init__()
+
+    def build(self, input_shape):
+        super(ScaleLayer, self).build(input_shape)
+
+    def call(self, X):
+        Y = tf.clip_by_value((((self.b - self.a)*( X - self.vars_min ) / (self.vars_max - self.vars_min)) + self.a), self.a, self.b)
```

Comment:
`(self.b - self.a) / (self.vars_max - self.vars_min)` can be computed once in the constructor, don't need to calc it for each call